cfme/cloud/tenant.py

```diff
@@ -101,12 +101,42 @@ class ProviderTenantAllView(TenantAllView):
 
     @property
     def is_displayed(self):
-        return (
-            self.logged_in_as_current_user and
-            self.navigation.currently_selected == ['Compute', 'Clouds', 'Providers'] and
-            self.entities.title.text == '{} (All Cloud Tenants)'.format(
-                self.context['object'].name)
-        )
+        def get_provider_navigation(provider):
+            # Return navigation path based on a type of a provider
+            from cfme.networks.provider import NetworkProvider
+            from cfme.cloud.provider import CloudProvider
+            if getattr(provider, 'name', False) and isinstance(provider, CloudProvider):
+                return ['Compute', 'Infrastructure', 'Providers']
+            elif getattr(provider, 'name', False) and isinstance(provider, NetworkProvider):
+                return ['Networks', 'Providers']
+            return []
+
+        expected_title = '{} (All Cloud Tenants)'
+        obj = self.context['object']
+        is_entity = getattr(obj, 'name', False) and isinstance(obj, BaseEntity)
```

Comment:
I'd prefer if we do it the same way as in L117:

```python
is_entity = isinstance(obj, BaseEntity) and obj.name
```

Reply:
The problem here is that `name` attribute does not exists in the definition of `BaseEntity` and this could result in an error in some cases. While `filter` is in the definition of `BaseCollection`

Reply:
@aljazkosir yep, that's exactly why I structured these two checks in this way in some of the other all methods :+1: